auth/loginHandler.js

```diff
@@ -0,0 +1,26 @@
+const bcrypt = require("bcrypt");
+
+const { getUserByEmail } = require("../controllers/users");
+const issueToken = require("./issueToken");
+
+const loginHandler = async (email, incomingPassword) => {
+  const user = await getUserByEmail(email);
+
+  if (!user) {
+    // eslint-disable-next-line no-throw-literal
```

Comment:
To taki średni pomysł, jak już chcesz rzucać błędami stąd to stworzył bym customowe błędy i potem je po constructorze rozróżniał 


```
class InvalidInputError extends Error {
  constructor(message) {
    super(message);
    this.name = 'InvalidInputError';
  }
}

class NotFoundError extends Error {
  constructor(message) {
    super(message);
    this.name = 'NotFoundError';
  }
}
```
```
  if (error1) {
    throw new NotFoundError(`Data with id ${id} not found.`);
  }

  if (error2) {
    throw new InvalidInputError('Invalid input. Expected a number.');
  }
```

```
try {
  await loginHandler(x,y);
} catch (error) {
  switch (error.constructor) {
      case InvalidInputError:
        // zwróć 400 na frontend
        break;
      case NotFoundError:
        // zwróć 404 na frontend
        break;
      default:
        console.error('Unknown Error:', error);
}
```